compiler-rt/test/builtins/Unit/sme-string-test.c

```diff
@@ -0,0 +1,183 @@
+// REQUIRES: linux, aarch64-target-arch
+// RUN: %clang_builtins %s %librt -o %t && %run %t
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#define N 1024
+#define NREPS 1234
```

Comment:
It looks like this test is testing more than just functionality (i.e. it seems to be testing performance). This can be seen from e.g. NREPS and `get_time_diff`. I think you'll want to make this purely a functional test and try to test out specific edge cases for each of the routines.

Reply:
Done.